### ISSUE TITLE
SKSE/Interfaces.h includes Windows.h

### ISSUE THREAD

**Issue by Ryan-rsm-McKenzie** [CharmedBaryon/CommonLibSSE-NG#58]
Why do you include `<Windows.h>` in a public header here? I don't see what's blocking you from putting this into a source file instead.
https://github.com/CharmedBaryon/CommonLibSSE-NG/blob/7bc34b175c33b4c96006967e683114fd69f0365b/include/SKSE/Interfaces.h#L12
`SKSE/Interfaces.h` is intended to be included throughout a developer's entire project, which means `Windows.h` will also be included throughout the entire project, and it is well known to not play nice with other libraries.

**Comment by Aietos** [comment]
I was just writing a similar issue when I noticed yours before submitting mine. Talk about timing!

Basically in my code, I need to check if an actor has the keyword NPC. Using this code, 

```
auto dobj = RE::BGSDefaultObjectManager::GetSingleton();
auto keywordNPC = dobj->GetObject<RE::BGSKeyword>(RE::DefaultObjectID::kKeywordNPC);

if (actor->HasKeyword(keywordNPC ) {
...
}
```

I get `class "RE::BGSDefaultObjectManager" has no member "GetObjectW"`. Precisely because `Windows.h` is overriding it.

In my PCH.h file, the order of the includes has to be like this:

```
#include "RE/Skyrim.h"
#include "SKSE/SKSE.h"
```

I tried adding `#undef GetObject`, but while it the code does compile, I get a CTD when `auto keywordNPC = dobj->GetObject<RE::BGSKeyword>(RE::DefaultObjectID::kKeywordNPC);` is called.

I'm not sure if the latter problem is related to this issue of `Windows.h`, but I can't think of a workaround for what I'm trying to do.

**Comment by jpstewart** [comment]
Resolved in 025ee60.
